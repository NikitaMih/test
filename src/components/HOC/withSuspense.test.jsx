import TestRenderer from 'react-test-renderer';
import React, { lazy } from 'react';
import withSuspense from './withSuspense';

const TestComponent = () => {
  return (
    <div>
      Компонент 1
    </div>
  );
};

const LazyTestComponent = lazy(() => TestComponent);

const MemoizedComponent = React.memo(() => {
  return (
    <div>
      <LazyTestComponent />
    </div>
  );
});

describe('withSuspence', () => {
  it('rendered lazy', async () => {
    const root = await TestRenderer.create(
      withSuspense(MemoizedComponent)
    );

    const tree = root.toJSON();

    /* await TestComponent; */
    expect(tree).toMatchSnapshot();
  });
});

