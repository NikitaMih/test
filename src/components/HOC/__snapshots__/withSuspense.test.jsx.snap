// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSuspence rendered lazy 1`] = `null`;
